translation/video.go

```diff
@@ -161,3 +204,204 @@ func (e *Engine) Text2Audio(content string) (*TTSResponse, error) {
 
 	return res, nil
 }
+
+// GenerateWebVTTFile generate caption for speech result
+func (e *Engine) GenerateWebVTTFile(asrTaskData ASRTaskData, path string) error {
+	var buffer bytes.Buffer
+
+	webVTTBuffer := GenerateSimpleWebVTTFileFromASRTaskData(asrTaskData)
+	_, err := webVTTBuffer.ToBuffer().WriteTo(&buffer)
+	if err != nil {
+		return err
+	}
+
+	// Write buffer to file
+	file, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = buffer.WriteTo(file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GenerateWebVTTFileWithTranslation generate caption for speech result with translation
+func (e *Engine) GenerateWebVTTFileWithTranslation(asrTaskData ASRTaskData, path string, from, to language.Tag) error {
+	var buffer bytes.Buffer
+
+	// Write head
+	webVTTBuffer := GenerateSimpleWebVTTFileFromASRTaskData(asrTaskData)
+
+	// Translate buffer
+	err := e.TranslateWebVTT(&webVTTBuffer, from, to)
+	if err != nil {
+		return err
+	}
+	_, err = webVTTBuffer.ToBuffer().WriteTo(&buffer)
+	if err != nil {
+		return err
+	}
+
+	// Write buffer to file
+	file, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = buffer.WriteTo(file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+const (
+	WebVTTFileHead      = "WEBVTT FILE\n\n"
+	WebVTTContentFormat = "%d\n%s --> %s\n%s\n\n"
+)
+
+// SimpleWebVTT is the simple format of WebVTT
+type SimpleWebVTT struct {
+	// Basic format
+	// WEBVTT FILE
+	//
+	// 1
+	// 00:00:09.500 --> 00:00:12.000
+	// The ocean floor rises 5 miles to the shores
+
+	WebVTTFileHead string
+	Content        []SimpleWebVTTContent
+}
+
+// SimpleWebVTTContent is the content of SimpleWebVTT
+type SimpleWebVTTContent struct {
+	Index     int
+	StartTime string
+	EndTime   string
+	Sentence  string
+}
+
+// NewSimpleWebVTT create a new SimpleWebVTT from io.Reader
+//
+// Example:
+// Basic format
+// WEBVTT FILE
+//
+// 1
+// 00:00:09.500 --> 00:00:12.000
+// The ocean floor rises 5 miles to the shores
+func NewSimpleWebVTT(reader io.Reader) SimpleWebVTT {
+	var simpleWebVTT SimpleWebVTT
+	var scanner = bufio.NewScanner(reader)
+	var line string
+	var lineNum int
+
+	// Skip first line
+	for scanner.Scan() {
+		line = scanner.Text()
+		lineNum++
+
+		// Check head
+		if fs := strings.Fields(line); len(fs) > 0 && fs[0] == "WEBVTT" {
+			simpleWebVTT.WebVTTFileHead = WebVTTFileHead
+			break
+		}
+	}
+
+	// Read content
+	for scanner.Scan() {
+		line = scanner.Text()
+		lineNum++
+
+		// Skip empty line
+		if line == "" {
+			continue
+		}
+
+		// Read batch content
+		if fs := strings.Fields(line); len(fs) > 0 {
+			// Read index
+			index, err := strconv.Atoi(fs[0])
+			if err != nil {
```

Comment:
recommend to add log here in order to record exception

Reply:
I try to throw an error in this funciton.